### PR TITLE
AS-4155: implement new salesforce nationality topology

### PR DESCRIPTION
**Pull request by rahul-r-jain**
SF Nationality Object schema (provided by Dani!) The ones with red accent are populated by SF itself, rest of the fields need to be populated via new nationality topology.

image

### CODE REVIEW

File: nm-salesforce-nationalities-topology/src/main/java/com/nutmeg/salesforce/nationalities/stream/joiner/CpdUserNationalityJoiner.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                .map(nationality -> mapToUserNationality(minimalUser, nationality))
+                .collect(Collectors.toList());
+        return UserNationalities.newBuilder()
+                .setUserUuid(minimalUser.getUuid())
+                .setProfileUuid(minimalUser.getProfileUUID())
+                .setNationalities(userNationalityList)
+                .build();
+    }
+
+    private UserNationality mapToUserNationality(MinimalUser minimalUser, CpdNationality cpdNationality) {
```
Comment (yu-nutmeg) [inline nm-salesforce-nationalities-topology/src/main/java/com/nutmeg/salesforce/nationalities/stream/joiner/CpdUserNationalityJoiner.java:28]: looks like we could use `mapstruct` here but I'm ok with doing it this way

Comment (rahul-r-jain) [reply nm-salesforce-nationalities-topology/src/main/java/com/nutmeg/salesforce/nationalities/stream/joiner/CpdUserNationalityJoiner.java:28]: sure thing.. will update using `mapstruct`

Comment (rahul-r-jain) [reply nm-salesforce-nationalities-topology/src/main/java/com/nutmeg/salesforce/nationalities/stream/joiner/CpdUserNationalityJoiner.java]: I've updated all joiners/mappers with `mapstruct` wherever possible

File: nm-salesforce-nationalities-avro/src/main/resources/avro/user-nationality/personal-details-nationality-list-by-user-uuid.avdl [hunk 2]
```diff
@@ -0,0 +1,17 @@
+@namespace("com.nutmeg.salesforce.avro.nationality")
+
+protocol AvroProtocol {
+
+  record PersonalDetailsNationalityList {
```
Comment (yu-nutmeg) [inline nm-salesforce-nationalities-avro/src/main/resources/avro/user-nationality/personal-details-nationality-list-by-user-uuid.avdl:5]: just make sure that you double check with Dani that the fields in the object you're sending to SF match those defined in the SF object

Comment (rahul-r-jain) [reply nm-salesforce-nationalities-avro/src/main/resources/avro/user-nationality/personal-details-nationality-list-by-user-uuid.avdl:5]: sure thing. I have confirmed with Dani on the SF nationality object schema and attached in the PR description for reference.

File: tools/k8s/prod.yaml [hunk 3]
```diff
@@ -56,6 +55,27 @@ environment:
...
+      replicas: 3
+      config:
+        retention.ms: "-1"
+        cleanup.policy: "delete"
+    - name: "com.nutmeg.crm.salesforce.user-nationality"
+      partitions: 8
+      replicas: 3
+      config:
+        retention.ms: "-1"
+        cleanup.policy: "delete"
```
[the remark this answers is not in the document]
Comment (rahul-r-jain) [reply tools/k8s/prod.yaml:77]: I checked `nm-customer-jurisdictions-service`, `nm-salesforce-kyc-aml-service` and existing `nm-salesforce-nationality-details` topologies and all of them use `cleanup.policy: "delete"`

Having said that, I'm happy to update it to `compact` but was trying to keep consistency among topologies.

**Review by yu-nutmeg — APPROVED** [review]

**Review by yu-nutmeg — APPROVED** [review]

**Review by iagoquirino — APPROVED** [review]

**Comment by iagoquirino** [comment]
Will the NINO check to send a flag if the nationality is valid or not will be part of this scope or tax residency one?

**Review by iagoquirino — APPROVED** [review]

**Pull request merged**